File: Go/PollingSystem2-go/models/vote.go
package models

import "time"

type Vote struct {
	PollID      int
	Option      string
	UserID      int
	SubmittedAt int64
}

func NewVote(pollID int, option string, userID int) (*Vote, error) {
	if pollID == 0 || option == "" || userID == 0 {
		return nil, ErrInvalidVote
	}
	return &Vote{
		PollID:      pollID,
		Option:      option,
		UserID:      userID,
		SubmittedAt: time.Now().UnixMilli(),
	}, nil
}

File: JavaScript/PollingSystem2/Vote.js
class Vote {
  pollId = 0;
  option = '';
  userId = 0;
  submittedAt = 0;

  constructor(pollId, option, userId) {
    if (!pollId || !option || !userId) {
      throw new Error('Invalid vote parameters!');
    }

    this.pollId = pollId;
    this.option = option;
    this.userId = userId;
    this.submittedAt = Date.now();
  }
}

module.exports = { Vote };
